reactor-core/src/main/java/reactor/core/scheduler/SchedulerMetricDecorator.java

```diff
@@ -78,7 +79,32 @@ in order to avoid consider two calls in a row as duplicates (yet still being abl
 		to distinguish between two instances with the same name and configuration).
 		 */
 
-		return ExecutorServiceMetrics.monitor(globalRegistry, service, executorId, tags);
+		class MetricsRemovingScheduledExecutorService extends DelegatingScheduledExecutorService {
+
+			MetricsRemovingScheduledExecutorService() {
+				super(ExecutorServiceMetrics.monitor(globalRegistry, service, executorId, tags));
+			}
+
+			@Override
+			public List<Runnable> shutdownNow() {
+				removeMetrics();
+				return super.shutdownNow();
+			}
+
+			@Override
+			public void shutdown() {
+				removeMetrics();
+				super.shutdown();
+			}
+
+			void removeMetrics() {
+				Search.in(globalRegistry)
+				      .tags(tags)
```

Comment:
Is the `executorId` already part of the `tags`? otherwise unrelated executor meters could be removed when a single executor is shutdown out of a pool.

Reply:
you're right. Fixing (and will add a test)